build-system/tasks/e2e/index.js

```diff
@@ -86,14 +86,23 @@ function createMocha_() {
     reporter = dotsReporter;
   }
 
-  return new Mocha({
+  const options = {
     // e2e tests have a different standard for when a test is too slow,
     // so we set a non-default threshold.
     slow: SLOW_TEST_THRESHOLD_MS,
     reporter,
     retries: TEST_RETRIES,
     fullStackTrace: true,
-  });
+  };
+
+  if (isCircleciBuild()) {
+    options.reporter = 'mocha-junit-reporter';
```

Comment:
This will overwrite the existing reporter, which is something we don't want. 

https://github.com/ampproject/amphtml/blob/c3ffc34af1200ac10ed41ea06a12ea6835c8a4e3/build-system/tasks/e2e/index.js#L81-L87

Since Mocha doesn't appear to support multiple reporters (https://github.com/mochajs/mocha/pull/1360), we may have to do a little surgery to combine this with one or more of the existing reporters. Our base reporter (written by Esther) is `dotsReporter`, which mimics the Karma reporter used by our unit and integration tests. `ciReporter` (written by Ricardo who was with us last summer) expands on `dotsReporter` and generates a report with test case stats. I'd imagine you can modify it to also generate the output needed here, or potentially replace it in future.

And if this turns out to be too complicated to do here, feel free to deal with the E2E test stuff in a separate PR.

Reply:
I was aware this would overwrite the reporter but wanted to make sure circleci would pick up the build results correctly BEFORE doing the surgery.